spring-integration-core/src/test/java/org/springframework/integration/config/annotation/MessagingAnnotationsWithBeanAnnotationTests.java

```diff
@@ -145,6 +150,8 @@ public void testMessagingAnnotationsFlow() {
 		assertNull(this.skippedChannel);
 		assertNull(this.skippedChannel2);
 		assertNull(this.skippedMessageSource);
+
+		assertNotNull(this.publisherChannel.receive(10000));
```

Comment:
Maybe it would help to check that the message received on the publisher channel is the one we expect? I know that this would be in a way re-testing the `@Publisher` annotation, I think of it more as defense against someone inadvertently altering the tests in the future - e.g. adding a second `@Publisher` test and so on.